src/index.js

```diff
@@ -101,12 +128,24 @@ export default class PreloadPlugin extends Plugin {
 				return;
 			}
 
-			fetch(
+			/**
+			 * If there is still another low-priority preloadRequest running,
+			 * abort it to save resources on the server.
+			 */
+			if (this.preloadRequest?.priority === 'low') {
+				this.preloadRequest.onreadystatechange = null;
+				this.preloadRequest.abort();
+				this.preloadRequest = null;
+			}
+
+			this.preloadRequest = fetch(
```

Comment:
Why are we using a new property `this.preloadRequest` instead of re-using `swup.preloadPromise`? Aren't these the same object? Or would we somehow lose access to it by not storing a copy in the plugin?

Reply:
The preloadPromise is not the actual request. We need to keep a reference to that one to be able to abort it.

Reply:
...of course we could actually put a reference to the request into the `preloadPromise` instead of the plugin instance, but it felt a bit overkill, refactor-wise.

Reply:
Oh, right.